explorer/explorerroutes.go

```diff
@@ -567,6 +569,14 @@ func (exp *explorerUI) timeBasedBlocksListing(val string, w http.ResponseWriter,
 
 // Blocks is the page handler for the "/blocks" path.
 func (exp *explorerUI) Blocks(w http.ResponseWriter, r *http.Request) {
+	bestBlockHeight, err := exp.dataSource.GetHeight()
```

Comment:
This should be moved after parsing query parameters so we don't hit the DB on invalid params.

Reply:
This particular method doesn't hit the DB. `GetHeight()`, which actually just calls `Height()`, does return the cached best block.  The `HeightDB` method on the other hand would hit the DB.